lib/board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+  attr_accessor :cells
+
+  @cells = Array.new(9, " ")
+
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  #-----puts out board with positions-----#
+  def initial_display
+    puts " 1 | 2 | 3 "
+    puts "-----------"
+    puts " 4 | 5 | 6 "
+    puts "-----------"
+    puts " 7 | 8 | 9 "
+  end
+
+  #-----prints out board----#
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  #-----returns the value of a position----#
+  def position(num)
+    cells[num.to_i - 1]
+  end
+
+  #-----checks for open spaces/" "-----#
+  def full?
+    !cells.include?(" ")
+  end
+
+  #-----counts cells containing markers-----#
+  def turn_count
+    cells.select {|m| m != " "}.count
+  end
+
+  #-----checks numbered position for !" "-----#
+  def taken?(num)
+    cells[num.to_i - 1] == "X" || cells[num.to_i - 1] == "O" ? true : false
```

Comment:
Avoid using the ternary operator to return `true` and `false`. Instead, rely on the implicit boolean return value of the expression. 

def taken?(num)
  cells[num.to_i - 1] == "X" || cells[num.to_i - 1] == "O"
end
```